frontend/apps/reader/modules/readerview.lua

```diff
@@ -487,6 +493,57 @@ function ReaderView:drawTempHighlight(bb, x, y)
 end
 
 function ReaderView:drawSavedHighlight(bb, x, y)
+    -- set/free note mark sign
+    if self.highlight.note_mark == "sidemark" then
```

Comment:
May be put this whole new chunk in its own method - and just call this new method here ? So it looks a little less scary :)

As I get it, this whole chunk is run on each page drawing/painting - but it looks that nothing depends on the page itself or its content. So, may be it (calling this new method) should just be elsewhere, in `ReaderView:recalculate()` maybe ?